apps/emqx_connector/src/emqx_connector.erl

```diff
@@ -458,3 +468,51 @@ ensure_no_channels(Configs) ->
         {error, Reason, _State} ->
             {error, Reason}
     end.
+
+to_bin(A) when is_atom(A) -> atom_to_binary(A, utf8);
+to_bin(B) when is_binary(B) -> B.
+
+validate_connector_name(ConnectorName) ->
+    try
+        _ = emqx_resource:validate_name(to_bin(ConnectorName)),
+        ok
+    catch
+        throw:Error ->
+            {error, Error}
+    end.
+
+validate_connector_name_in_config(Path) ->
+    case Path of
+        [?ROOT_KEY, _ConnectorType, ConnectorName] ->
+            validate_connector_name(ConnectorName);
+        _ ->
```

Comment:
Is this something to expect? When? Otherwise, wouldn't it be better to just crash?

Reply:
True, I don't think this should be reached.  I'll removed it.

Reply:
Actually, no, it may happen if someone does `emqx_conf:update([connectors, type], ...)`.